idea/data-storage/src/database/entities/state.entity.ts

```diff
@@ -0,0 +1,28 @@
+import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+import { IState } from '@gear-js/common';
+
+import { Code } from './code.entity';
+
+@Entity()
+export class State implements IState {
+  @PrimaryGeneratedColumn('uuid')
+  public id: string;
+
+  @Column()
+  public name: string;
+
+  @Column()
+  public wasmBuffBase64: string;
+
+  @Column({ type: 'json' })
+  public funcNames: string;
+
+  @Column({ type: 'json' })
+  public functions: string;
```

Comment:
What is the diferrence between these fields?

Reply:
Include this field for making able to do query requests with a `function name`. 

Before including this field I tested it, it is possible to make a query request by `function` name when in our field `functions` will have a structure like the below.

```javascript
State { functions : { func_1: some_value, func_2: some_value, func_3: some_value }}
```

Query request didn't work if we want to get by a key of the `object` but by `value` of key, it works.
Example: 

 #### Search by key in JSON field
  🔴 Didn't work  
```javascript
 Object { [search_by_key]: some_value } 
 ``` 
 #### Search by value in JSON field
 ✅ Works  
  ``` javascript 
  Object { key: search_by_value } or { key: [func_1, func_2, func_3] }
  ```